packages/phishing-controller/src/PhishingDetector.ts

```diff
@@ -0,0 +1,177 @@
+import { distance } from 'fastest-levenshtein';
+
+import {
+  domainPartsToDomain,
+  domainPartsToFuzzyForm,
+  domainToParts,
+  getDefaultPhishingDetectorConfig,
+  matchPartsAgainstList,
+  processConfigs,
+} from './utils';
+
+export type LegacyPhishingDetectorList = {
+  whitelist?: string[];
+  blacklist?: string[];
+} & FuzzyTolerance;
+
+export type PhishingDetectorList = {
+  allowlist?: string[];
+  blocklist?: string[];
+  name?: string;
+  version?: string | number;
+} & FuzzyTolerance;
+
+export type FuzzyTolerance =
+  | {
+      tolerance?: number;
+      fuzzylist: string[];
+    }
+  | {
+      tolerance?: never;
+      fuzzylist?: never;
+    };
+
+export type PhishingDetectorOptions =
+  | LegacyPhishingDetectorList
+  | PhishingDetectorList[];
+
+export type PhishingDetectorConfiguration = {
+  name?: string;
+  version?: number | string;
+  allowlist: string[][];
+  blocklist: string[][];
+  fuzzylist: string[][];
+  tolerance: number;
+};
+
+export type PhishingDetectorResult = {
+  name?: string;
+  version?: string;
+  result: boolean;
+  match?: string;
+  type: 'all' | 'fuzzy' | 'blocklist' | 'allowlist' | 'blacklist' | 'whitelist';
+};
+
+export class PhishingDetector {
+  #configs: PhishingDetectorConfiguration[];
+
+  #legacyConfig: boolean;
```

Comment:
These two were originally public in `eth-phishing-detector`, but looks like they are not used outside of this class - we should be able to prepend # to these without issues